x/emissions/metrics/metrics_util.go

```diff
@@ -0,0 +1,199 @@
+		[]string{"allora", "websocket", "connect"},
+		1,
+		nil,
+	)
+}
+
+// Measures throughput per message type
+// Metric Name:
+//
+//	allora_throughput_<metric_name>
+func SetThroughputMetricByType(metricName string, value float32, msgType string) {
+	telemetry.SetGaugeWithLabels(
+		[]string{"allora", "loadtest", "tps", metricName},
+		value,
+		[]metrics.Label{telemetry.NewLabel("msg_type", msgType)},
+	)
+}
+
+// Measures the number of times the total block gas wanted in the proposal exceeds the max
+// Metric Name:
+//
+//	allora_failed_total_gas_wanted_check
+func IncrFailedTotalGasWantedCheck(proposer string) {
+	telemetry.IncrCounterWithLabels(
+		[]string{"allora", "failed", "total", "gas", "wanted", "check"},
+		1,
+		[]metrics.Label{telemetry.NewLabel("proposer", proposer)},
+	)
+}
+
+// Measures the number of times the total block gas wanted in the proposal exceeds the max
+// Metric Name:
+//
+//	allora_failed_total_gas_wanted_check
+func IncrValidatorSlashed(proposer string) {
+	telemetry.IncrCounterWithLabels(
+		[]string{"allora", "failed", "total", "gas", "wanted", "check"},
+		1,
+		[]metrics.Label{telemetry.NewLabel("proposer", proposer)},
+	)
+}
+
+// Measures the number of times the total block gas wanted in the proposal exceeds the max
+// Metric Name:
+//
+//	allora_tx_gas_counter
+func IncrGasCounter(gasType string, value int64) {
+	telemetry.IncrCounterWithLabels(
+		[]string{"allora", "tx", "gas", "counter"},
+		float32(value),
+		[]metrics.Label{telemetry.NewLabel("type", gasType)},
+	)
+}
+
+// Measures RPC endpoint request throughput
+// Metric Name:
+//
+//	allora_rpc_request_counter
+func IncrementRpcRequestCounter(endpoint string, connectionType string, success bool) {
+	telemetry.IncrCounterWithLabels(
+		[]string{"allora", "rpc", "request", "counter"},
+		float32(1),
+		[]metrics.Label{
+			telemetry.NewLabel("endpoint", endpoint),
+			telemetry.NewLabel("connection", connectionType),
+			telemetry.NewLabel("success", strconv.FormatBool(success)),
+		},
+	)
+}
+
+// Measures the RPC request latency in milliseconds
+// Metric Name:
+//
+//	allora_rpc_request_latency_ms
+func MeasureRpcRequestLatency(endpoint string, connectionType string, startTime time.Time) {
+	metrics.MeasureSinceWithLabels(
+		[]string{"allora", "rpc", "request", "latency_ms"},
+		startTime.UTC(),
+		[]metrics.Label{
+			telemetry.NewLabel("endpoint", endpoint),
+			telemetry.NewLabel("connection", connectionType),
+		},
+	)
+}
+
+// IncrProducerEventCount increments the counter for events produced.
+// This metric counts the number of events produced by the system.
+// Metric Name:
+//
+//	allora_loadtest_produce_count
+func IncrProducerEventCount(msgType string) {
+	telemetry.IncrCounterWithLabels(
+		[]string{"allora", "loadtest", "produce", "count"},
+		1,
+		[]metrics.Label{telemetry.NewLabel("msg_type", msgType)},
+	)
+}
+
+func RecordMetrics(apiMethod string, connectionType string, startTime time.Time, success bool) {
```

Comment:
I don't think the `connectionType` is never not `"rpc"` in your PR here, nor do I think that this code knows if it's being invoked via RPC or API.

=> I'd remove it

If it is kept, I'd make it an enum. I don't know what the other values would be though because of what I just said